includes/class-newspack-popups-inserter.php

```diff
@@ -1057,6 +1057,19 @@ public static function should_display( $popup, $check_if_is_post = false ) {
 		}
 		$is_post_context_matching = $is_taxonomy_matching && in_array( $post_type, $supported_post_types );
 
+		if ( $is_post_context_matching ) {
+			/**
+			 * Allow additional checks to be performed before displaying a prompt.
+			 *
+			 * Additional checks will be checked alongside regular checks for taxonomies and post types, but will not override them. All checks must pass for the prompt to be displayed.
```

Comment:
> Can you provide an example of this filter being used?

https://github.com/Automattic/newspack-multibranded-site/blob/master/includes/customizations/class-popups-should-display-prompt.php

> What alongside check is being performed? ... It's also not clear to me why the filter is only applied when the value is true?

My rationale is that we don't want plugins to completely override the checks that happen here. For example, if I configured a prompt to only display for a certain post type, I don't want plugins changing that, because that would break the user experience and what they expect from the plugin.

So, in practice, the only thing plugins can do is keep the prompts from being displayed based on their rules.

Another way to write it was like this:

```PHP
$additional_checks = apply_filters( 'newspack_popups_should_display_prompt_additional_checks', true, $popup, $check_if_is_post );

return $is_post_context_matching && $additional_checks;
```

But the `if true` saves us a the work of doing useless check if the previous checks already returned false.

Reply:
I like the idea of having the `$additional_checks` better because it adds extra intention to what you're describing.

A non-blocking thought: Although I can't think of an example right now I still think that it could also simply be a generic filter for the return of the method. There are several instances in which WP applies filters to core aspects of its functionality that you could break everything if poorly used. Being in code, I don't think it's for us to decide how/why to filter a value.

Reply:
> I like the idea of having the $additional_checks better

Yes, it's more readable. But I think we should keep the `if` to avoind unnecessary checks.

Or we can change to just a filter that filters it all. I tend to agree with you but I also think we could wait for a usecase to add that filter... it feels like nothing but it's additional processing for each prompt all the time...

Not a strong opinion though

Reply:
Liked your last suggestion. But ok, for the sake of simplicity let's filter the whole thing